### PR TITLE
Elaborate on Python support policy

### LINKED ISSUE #2251: [Python 2 has been deprecated] - Formalise Python support policy
### Status update

_November 1st, 2021 @ 10:36AM EST_ by @ichard26 

**We've deprecated Python 2 support starting with version 21.10b0**. Assuming things go as planned the support will be **dropped sometime in January of 2022** along side the first stable release of Black.

---

From the discussions in #2247. The new FAQ (once merged) contains information on the supported Python versions. @ichard26 mentioned that it would be nice to have such documentation elsewhere in the docs too.

As discussed, the support plan seems to essentially be: "support whatever CPython versions are currently not EOL'd". This is for running Black. Then another question is formatting files, which is currently also done for 2.7.

- Should the Python version support of Black be considered identical to CPython official support?
- Does file formatting follow that, or do we intend to support older versions too?

### PR DESCRIPTION
**Pull request by felix-hilden**
### Description
Closes #2251: I've included a short Python support policy to our FAQ. TL;DR: we promise to support all non-EOL versions, but don't promise to keep old 3+ versions laying around if we decide to remove them at some point.

### Checklist - did you ...

- [x] Add new / update outdated documentation?

### Some discussion:
- Is the content fine? It's basically what we do now, or at least the least strict and most convenient version of it. We could promise a deprecation period as well.
- Should we relocate the text? The Python 2 support text was nice as a FAQ entry, but this could be better off somewhere else.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ and the first release covered by our new stability policy.
 - Change HTML theme to Furo primarily for its responsive design and mobile support
   (#2793)
 - Deprecate the `black-primer` tool (#2809)
+- Document Python support policy (#2819)
```
Comment (ichard26) [inline CHANGES.md:87]: Arguably this could count as a highlight

**Review by cooperlees — APPROVED** [review]
Nice. Agree with it all. Thanks!

File: CHANGES.md (hunk 1 above)
Comment (felix-hilden) [reply CHANGES.md:87]: I hesitated, because it doesn't really affect anything currently 😄 but I can change it if you'd like

Comment (cooperlees) [reply CHANGES.md:87]: Can't hurt to have ti here ... might make more people come read it advertising it here ...

Comment (felix-hilden) [reply CHANGES.md:87]: Not sure if I misunderstood you Cooper, but I think Richard was asking if it should be put in the "highlights" section at the very top instead

File: docs/faq.md [hunk 2]
```diff
@@ -71,9 +71,13 @@ readability because operators are misaligned. Disable W503 and enable the
 disabled-by-default counterpart W504. E203 should be disabled while changes are still
 [discussed](https://github.com/PyCQA/pycodestyle/issues/373).
 
-## Does Black support Python 2?
+## Which Python versions does Black support?
 
-Support for formatting Python 2 code was removed in version 22.0.
+_Black_ supports at least all Python versions that have not reached their end of life.
```
Comment (JelleZijlstra) [inline docs/faq.md:76]: This FAQ entry doesn't actually answer the question: we currently support 3.6 through 3.10.

"Supporting" a Python version in Black can mean two things: running Black under that version, or formatting all code that's valid for that version. Those two meanings are the same right now, but they weren't in the past (we supported formatting Python 2 but not running under Python 2; we supported running under 3.10 before we could fully format 3.10) and they might not be again in the future. Maybe we can make this more explicit too.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**